### DISCUSSION TITLE
TypeScript support for undefined values

### DISCUSSION THREAD

**Discussion by sambecker** [porsager/postgres#868] [Q&A]
In order to accept `undefined` in queries, I've applied the following configuration:
```ts
export const sql = postgres(process.env.POSTGRES_URL, {
  transform: { undefined: null },
});
```

But then TypeScript says its usage is not valid:
```ts
// TS Error: 'ParameterOrFragment<never>'.ts(2769)
sql`UPDATE table SET field=${undefined} WHERE id=${1}`
```

Is there a recommended approach to loosen the types?

**Comment by insidewhy** [comment]
In the `types/index.d.ts` changing the following definition:

```typescript
type Serializable = never | null | boolean | number | string | Date | Uint8Array
```

To include `undefined` fixes this.

**Reply by insidewhy** [reply]
Wherever you use `Sql` you can use `Sql<Record<string, undefined>>` and wherever you use `postgres` you can use `postgres<Record<string, undefined>>` and then this seems to be fine.
